com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/resource/AbstractResourceDescriptionDelta.java

```diff
@@ -0,0 +1,52 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Avaloq Evolution AG and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Avaloq Evolution AG - initial API and implementation
+ *******************************************************************************/
+
+package com.avaloq.tools.ddk.xtext.resource;
+
+import java.util.Map;
+
+import org.eclipse.xtext.resource.IResourceDescription.Delta;
+
+import com.google.common.collect.Maps;
+
+
+/**
+ * Support for extensions on resource description delta.
+ */
+public abstract class AbstractResourceDescriptionDelta implements Delta {
+
+  private final Map<Class<? extends Object>, Object> extensions = Maps.newHashMap();

```

Comment:
I think I would probably allocate the HashMap on demand. An empty HashMap only uses around 128 bytes of memory, IIRC, but for 100K sources that would still add up to 12 MiB of memory.

Reply:
Can do that, but current implementation of the consumer will always add something for incremental builds. Only full builds will benefit. If this matters, I need to rethink consumer logic too.

Reply:
Why can't you just deal with this field being "null" in the instance methods of this class? Why would consumers have to change?

Reply:
OK. I see you already changed the code. That is what I meant.